### PR TITLE
Shared sampler for VT cache samples

### PR DESCRIPTION
**Pull request by robin-demoor**
### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [ ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Currently the VirtualTexturing Sample Node / VTProperty use 1 sampler per layer while the samplers are effectively the same.
This change makes VT Sampling only use 1 sampler by sharing the first sampler it encounters to all VT samples.

---
### Testing status
Describe what manual/automated tests were performed for this PR

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by sebastienlagarde** [comment]
added Chris as this PR touch shader graph

File: com.unity.shadergraph/Editor/Data/Graphs/VirtualTextureShaderProperty.cs [hunk 1]
```diff
@@ -100,8 +100,14 @@ internal override void ForeachHLSLProperty(Action<HLSLProperty> action)
 
                 Action<ShaderStringBuilder> customDecl = (builder) =>
                 {
-                    // declare texture stack
+                    //Declare sampler if not defined already, use the first layer to get the proper settings for VT textures.
+                    builder.AppendIndentation();
+                    builder.AppendLine("#ifndef VT_SAMPLER");
+                    builder.AppendLine("#define VT_SAMPLER " + "sampler" + referenceName + "_c0");
+                    builder.AppendLine("SAMPLER(sampler" + referenceName + "_c0);");
```
Comment (cdxntchou) [inline com.unity.shadergraph/Editor/Data/Graphs/VirtualTextureShaderProperty.cs:107]: We might be able to declare this sampler as an actual HLSLProperty (see code just above this).

I guess the question is whether it properly de-duplicates identical HLSLProperties, if it is multiply-defined.

What you have here would work, but may break hybrid compatibility if/when they introduce Bind Sets..  but we could also look to fix this code when that happens.

Comment (robin-demoor) [reply com.unity.shadergraph/Editor/Data/Graphs/VirtualTextureShaderProperty.cs]: It works if I use a custom declaration HLSL property.
https://github.com/Unity-Technologies/Graphics/pull/3538/commits/c564d658571feb1f4b5a5712447224292bf225b9

**Comment by robin-demoor** [comment]
closing PR as it is blocked by trunk bug https://fogbugz.unity3d.com/f/cases/1306556/

**Pull request closed without merging**